src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -75,33 +107,177 @@ public JwtVendor(final Settings settings, final Optional<LongSupplier> timeProvi
      *   PublicKeyUse: SIGN
      *   Encryption Algorithm: HS512
      * */
-    static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
-        String signingKey = settings.get("signing_key");
-
-        if (!Strings.isNullOrEmpty(signingKey)) {
+    static JsonWebKey createJwkFromSettings(Settings settings) {
+        String algorithm = settings.get("algorithm");
+        Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
 
+        if (algorithm != null) {
+            return getSigningJwk(algorithm.toUpperCase(Locale.ROOT), settings);
+        } else if (!jwkSettings.isEmpty()) {
             JsonWebKey jwk = new JsonWebKey();
 
-            jwk.setKeyType(KeyType.OCTET);
-            jwk.setAlgorithm("HS512");
-            jwk.setPublicKeyUse(PublicKeyUse.SIGN);
-            jwk.setProperty("k", signingKey);
+            for (String key : jwkSettings.keySet()) {
+                jwk.setProperty(key, jwkSettings.get(key));
+            }
 
             return jwk;
         } else {
-            Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
+            // try default
+            return getSigningJwk("HS512", settings);
+        }
+    }
+
+    private static JsonWebKey getSigningJwk(String algorithm, Settings settings) throws IllegalArgumentException {
```

Comment:
Nit; These static functions seem like they below in a new class that should be oriented towards building a JsonWebKey instance to reduce coupling with this specific key OnBehalfOf token vendor